test/e2e/node_feature_discovery_test.go

```diff
@@ -787,6 +787,24 @@ core:
 					By("Verfiying node status capacity from NodeFeatureRules #4")
 					eventuallyNonControlPlaneNodes(ctx, f.ClientSet).Should(MatchCapacity(expectedCapacity, nodes, false))
 
+					By("Creating NodeFeatureRules #5")
+					Expect(testutils.CreateNodeFeatureRulesFromFile(ctx, nfdClient, "nodefeaturerule-5.yaml")).NotTo(HaveOccurred())
+
+					By("Verifying node annotations from NodeFeatureRules #5")
+					expectedAnnotations["*"] = k8sAnnotations{
+						nfdv1alpha1.FeatureLabelNs + "/defaul-ns-annotation":   "foo",
+						nfdv1alpha1.FeatureLabelNs + "/defaul-ns-annotation-2": "bar",
+						"custom.vendor.example/feature":                        "baz",
+					}
+					eventuallyNonControlPlaneNodes(ctx, f.ClientSet).Should(MatchAnnotations(expectedAnnotations, nodes, true))
+
+					By("Deleting NodeFeatureRule object")
+					err = nfdClient.NfdV1alpha1().NodeFeatureRules().Delete(ctx, " e2e-feature-annotations-test", metav1.DeleteOptions{})
+					Expect(err).NotTo(HaveOccurred())
+
+					By("Verifying node annotations from NodeFeatureRules #5")
+					eventuallyNonControlPlaneNodes(ctx, f.ClientSet).Should(MatchAnnotations(expectedAnnotations, nodes, false))
```

Comment:
Hmm, this shouldn't work 🤔 The `expectedAnnotations` should basically be empty after deleting the CR